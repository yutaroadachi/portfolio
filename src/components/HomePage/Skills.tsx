import { Box, Center, chakra, ListItem, UnorderedList } from '@chakra-ui/react'
import React from 'react'

export const Skills = () => (
  <Box py={6}>
    <chakra.h2 textStyle="h2">Skills</chakra.h2>
    <Center>
      <UnorderedList>
        {skills.map((skill) => (
          <ListItem key={skill}>{skill}</ListItem>
        ))}
      </UnorderedList>
    </Center>
  </Box>
)

const skills = [
  'TypeScript',
  'React',
  'Next.js',
  'Chakra UI',
  'Ruby on Rails',
  'Go',
]
