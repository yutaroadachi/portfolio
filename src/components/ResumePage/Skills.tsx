import { Box, chakra, Tag, Wrap, WrapItem } from '@chakra-ui/react'
import React from 'react'

export type SkillsProps = {}

export const Skills = (props: SkillsProps) => (
  <Box>
    <chakra.h2 textStyle="h2WithTreat">Skills</chakra.h2>
    <Wrap>
      {skills.map(({ label, field }) => (
        <WrapItem key={label}>
          <Tag variant={field === 'frontend' ? 'primary' : 'outlined'}>
            {label}
          </Tag>
        </WrapItem>
      ))}
    </Wrap>
  </Box>
)

type Skill = {
  label: string
  field: 'frontend' | 'backend'
}

const skills: Skill[] = [
  { label: 'React', field: 'frontend' },
  { label: 'TypeScript', field: 'frontend' },
  { label: 'Next.js', field: 'frontend' },
  { label: 'Chakra UI', field: 'frontend' },
  { label: 'Context', field: 'frontend' },
  { label: 'SWR', field: 'frontend' },
  { label: 'Formik', field: 'frontend' },
  { label: 'Yup', field: 'frontend' },
  { label: 'Jest', field: 'frontend' },
  { label: 'Testing Library', field: 'frontend' },
  { label: 'Cypress', field: 'frontend' },
  { label: 'Storybook', field: 'frontend' },
  { label: 'Hygen', field: 'frontend' },
  { label: 'Ruby on Rails', field: 'backend' },
  { label: 'Go', field: 'backend' },
]
