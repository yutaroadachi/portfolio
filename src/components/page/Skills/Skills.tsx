import { FiExternalLink } from 'react-icons/fi'
import { Biography } from 'src/components/domain/Biography'

export const Skills = () => {
  return (
    <div className="space-y-8">
      <section>
        <h1 id="skills">スキル</h1>
      </section>
      <FrontEnd />
      <BackEnd />
      <Biography />
    </div>
  )
}

const FrontEnd = () => {
  return (
    <section className="space-y-4">
      <h2 className="h2-with-treat">フロントエンド</h2>
      <table className="table-auto w-full text-sm">
        <thead>
          <tr className="text-left whitespace-nowrap">
            <th className="p-4">技術</th>
            <th className="p-4">経験年数</th>
            <th className="p-4">詳細</th>
          </tr>
        </thead>
        <tbody>
          <tr className="text-left">
            <td className="whitespace-nowrap p-4">React</td>
            <td className="whitespace-nowrap p-4 ">3.5年</td>
            <td rowSpan={3} className="p-4">
              <div className="space-y-4">
                <div>
                  WebメディアのようなHeadless
                  CMSを使用したJamstackなWebサイトから、複雑なUIやロジックを持ったWebアプリケーションの開発まで対応可能です。また、CI/CDやStorybookの整備などフロントエンド開発基盤の構築も対応可能です。
                </div>
                <div>
                  また、
                  <a
                    href="https://zenn.dev/a_da_chi"
                    target="_blank"
                    rel="noopener noreferrer"
                    className="text-c-link mx-1"
                  >
                    Zenn
                    <FiExternalLink className="inline-block mb-1 ml-1" />
                  </a>
                  にて「
                  <a
                    href="https://zenn.dev/a_da_chi/articles/105dac5573b2f5"
                    target="_blank"
                    rel="noopener noreferrer"
                    className="text-c-link mx-1"
                  >
                    もう迷わないNext.jsのCSR/SSR/SSG/ISR
                    <FiExternalLink className="inline-block mb-1 ml-1" />
                  </a>
                  」などNext.jsに関する記事を複数執筆しており、多数のLikeを頂いています。
                </div>
                <div className="space-y-1">
                  <div>その他周辺ライブラリ</div>
                  <ul className="list-disc list-inside">
                    <li>バンドラー: Webpack / Vite</li>
                    <li>ルーティング: React Router</li>
                    <li>コンポーネント: Chakra UI / MUI</li>
                    <li>
                      CSS: Tailwind CSS / Bootstrap / CSS Modules / styled
                      components / Emotion
                    </li>
                    <li>状態管理: Context API / Jotai</li>
                    <li>データ取得: SWR</li>
                    <li>
                      GraphQL: Apollo Client / urql / GraphQL Code Generator
                    </li>
                    <li>フォーム: Formik / Yup</li>
                    <li>
                      テスト: Jest / Vitest / Testing Library / reg-suit /
                      Chromatic / Cypress
                    </li>
                    <li>
                      コンポーネントカタログ: Storybook / Storyshots / Storycap
                    </li>
                    <li>コードジェネレーター: Scaffdog / Hygen</li>
                  </ul>
                </div>
              </div>
            </td>
          </tr>
          <tr className="text-left">
            <td className="whitespace-nowrap p-4">TypeScript</td>
            <td className="whitespace-nowrap p-4 ">2.5年</td>
          </tr>
          <tr className="text-left">
            <td className="whitespace-nowrap p-4">Next.js</td>
            <td className="whitespace-nowrap p-4 ">2.5年</td>
          </tr>
        </tbody>
      </table>
    </section>
  )
}

const BackEnd = () => {
  return (
    <section className="space-y-4">
      <h2 className="h2-with-treat">バックエンド</h2>
      <table className="table-auto w-full text-sm">
        <thead>
          <tr className="text-left whitespace-nowrap">
            <th className="p-4">技術</th>
            <th className="p-4">経験年数</th>
            <th className="p-4">詳細</th>
          </tr>
        </thead>
        <tbody>
          <tr className="text-left">
            <td className="whitespace-nowrap p-4">Ruby on Rails</td>
            <td className="whitespace-nowrap p-4 ">3.5年</td>
            <td rowSpan={1} className="p-4">
              <div className="space-y-4">
                <div>
                  モノリシックなアプリケーションから、API（REST,
                  GraphQL）の開発まで対応可能です。
                </div>
                <div className="space-y-1">
                  <div>その他周辺ライブラリ</div>
                  <ul className="list-disc list-inside">
                    <li>GraphQL: GraphQL Ruby</li>
                  </ul>
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </section>
  )
}
