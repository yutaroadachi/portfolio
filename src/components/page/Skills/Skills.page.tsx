import React, { ComponentProps } from 'react'
import { Layout } from 'src/components/ui/Layout'
import { MetaTag } from 'src/components/ui/MetaTag'
import { Skills } from './Skills'

export type SkillsPageProps = ComponentProps<typeof Skills>

export const SkillsPage = ({}: SkillsPageProps) => {
  return (
    <Layout>
      <MetaTag
        title="Skills"
        description="Yutaro Adachi(フロントエンドエンジニア)のスキルです。"
      />
      <Skills />
    </Layout>
  )
}
