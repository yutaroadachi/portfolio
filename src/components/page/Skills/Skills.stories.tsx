import { Meta, Story } from '@storybook/react'
import { Skills } from './Skills'

export default {
  title: 'page/Skills',
  component: Skills,
} as Meta

type Template = Story

const Template: Template = (args) => <Skills {...args} />

export const Default: Template = Template.bind({})
