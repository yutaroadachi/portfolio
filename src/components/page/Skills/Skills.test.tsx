import React from 'react'
import { render } from 'src/__tests__/utils'
import { Skills } from './Skills'

test('Skills', () => {
  const { container } = render(<Skills />)
  expect(container).toMatchSnapshot()
})
