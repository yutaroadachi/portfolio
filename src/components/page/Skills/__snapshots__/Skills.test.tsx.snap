// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills 1`] = `
<div>
  <div
    class="css-0"
  >
    <h1
      class="css-15mzpqr"
      id="skills"
    >
      スキル
    </h1>
    <div
      class="chakra-stack css-aajx3r"
    >
      <div
        class="css-1ki54i"
      >
        <table
          class="chakra-table css-xys6pb"
          role="table"
        >
          <thead
            class="css-0"
          >
            <tr
              class="css-0"
              role="row"
            >
              <th
                class="css-iz806w"
              >
                技術
              </th>
              <th
                class="css-iz806w"
              >
                経験年数
              </th>
              <th
                class="css-1tk997h"
              >
                詳細
              </th>
            </tr>
          </thead>
          <tbody
            class="css-0"
          >
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-1s6o4al"
                role="gridcell"
              >
                React
              </td>
              <td
                class="css-1s6o4al"
                role="gridcell"
              >
                2.5年
              </td>
              <td
                class="css-nzu2gk"
                role="gridcell"
              >
                Hooksを使用して関数コンポーネントを宣言的に書くことができます。
                <ul
                  class="css-1mqw3so"
                  role="list"
                >
                  <li
                    class="css-0"
                  >
                    スタイリング：styled-components
                  </li>
                  <li
                    class="css-0"
                  >
                    UIライブラリ：Chakra UI/MUI
                  </li>
                  <li
                    class="css-0"
                  >
                    状態管理：Context API
                  </li>
                  <li
                    class="css-0"
                  >
                    データフェッチ：SWR
                  </li>
                  <li
                    class="css-0"
                  >
                    フォームライブラリ：Formik/Yup
                  </li>
                  <li
                    class="css-0"
                  >
                    単体テスト：Jest
                  </li>
                  <li
                    class="css-0"
                  >
                    結合テスト：Testing Library
                  </li>
                  <li
                    class="css-0"
                  >
                    E2E：Cypress
                  </li>
                  <li
                    class="css-0"
                  >
                    コンポーネントカタログ：Storybook
                  </li>
                  <li
                    class="css-0"
                  >
                    コードジェネレータ：Hygen
                  </li>
                </ul>
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-1s6o4al"
                role="gridcell"
              >
                TypeScript
              </td>
              <td
                class="css-1s6o4al"
                role="gridcell"
              >
                1.5年
              </td>
              <td
                class="css-nzu2gk"
                role="gridcell"
              >
                基本的な型付けからジェネリクスを用いた少し応用的な型付けまで対応可能です。
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-1s6o4al"
                role="gridcell"
              >
                Next.js
              </td>
              <td
                class="css-1s6o4al"
                role="gridcell"
              >
                1.5年
              </td>
              <td
                class="css-nzu2gk"
                role="gridcell"
              >
                <a
                  class="chakra-link css-wmzrah"
                  href="https://zenn.dev/a_da_chi/articles/181ea4ccc39580"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  基本的なNext.jsプロジェクトのセットアップ
                  <svg
                    class="chakra-icon css-1pmp75g"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-width="2"
                    >
                      <path
                        d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                      />
                      <path
                        d="M15 3h6v6"
                      />
                      <path
                        d="M10 14L21 3"
                      />
                    </g>
                  </svg>
                </a>
                から対応可能です。上記の記事以外にも
                <a
                  class="chakra-link css-wmzrah"
                  href="https://zenn.dev/a_da_chi"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Zenn
                  <svg
                    class="chakra-icon css-1pmp75g"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-width="2"
                    >
                      <path
                        d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                      />
                      <path
                        d="M15 3h6v6"
                      />
                      <path
                        d="M10 14L21 3"
                      />
                    </g>
                  </svg>
                </a>
                でNext.jsに関する記事を複数執筆しており、累計で500近くのLikeを頂いております。
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-1s6o4al"
                role="gridcell"
              >
                Ruby on Rails
              </td>
              <td
                class="css-1s6o4al"
                role="gridcell"
              >
                2.5年
              </td>
              <td
                class="css-nzu2gk"
                role="gridcell"
              >
                モノリシックなアプリケーション開発からAPI開発まで対応可能です。Fat Controllerにならないようにするのはもちろん、デコレータ層やユースケース層を使用してFat Modelにならないように意識しています。テストにはminitestを使用することが多く、日常的にテストを書いています。
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-1s6o4al"
                role="gridcell"
              >
                Go
              </td>
              <td
                class="css-1s6o4al"
                role="gridcell"
              >
                0.1年
              </td>
              <td
                class="css-nzu2gk"
                role="gridcell"
              >
                社内ツールの作成で触ったことがある程度です。
              </td>
            </tr>
          </tbody>
        </table>
        <div
          class="chakra-accordion css-259i0c"
        >
          <div
            class="chakra-stack css-egoftb"
          >
            <div
              class="chakra-accordion__item css-ejn068"
            >
              <button
                aria-controls="accordion-panel-1"
                aria-expanded="true"
                class="chakra-accordion__button css-df84mh"
                data-index="0"
                id="accordion-button-1"
                type="button"
              >
                <div
                  class="css-1eziwv"
                >
                  <h3
                    class="css-1vpdv9u"
                  >
                    React
                  </h3>
                  <p
                    class="css-1vf9ext"
                  >
                    2.5年
                  </p>
                </div>
                <svg
                  aria-hidden="true"
                  class="chakra-icon chakra-accordion__icon css-j2ph2z"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <div
                class="chakra-collapse"
                style="overflow: hidden; display: block; opacity: 0; height: auto;"
              >
                <div
                  aria-labelledby="accordion-button-1"
                  class="chakra-accordion__panel css-1hb5ode"
                  id="accordion-panel-1"
                  role="region"
                >
                  Hooksを使用して関数コンポーネントを宣言的に書くことができます。
                  <ul
                    class="css-1mqw3so"
                    role="list"
                  >
                    <li
                      class="css-0"
                    >
                      スタイリング：styled-components
                    </li>
                    <li
                      class="css-0"
                    >
                      UIライブラリ：Chakra UI/MUI
                    </li>
                    <li
                      class="css-0"
                    >
                      状態管理：Context API
                    </li>
                    <li
                      class="css-0"
                    >
                      データフェッチ：SWR
                    </li>
                    <li
                      class="css-0"
                    >
                      フォームライブラリ：Formik/Yup
                    </li>
                    <li
                      class="css-0"
                    >
                      単体テスト：Jest
                    </li>
                    <li
                      class="css-0"
                    >
                      結合テスト：Testing Library
                    </li>
                    <li
                      class="css-0"
                    >
                      E2E：Cypress
                    </li>
                    <li
                      class="css-0"
                    >
                      コンポーネントカタログ：Storybook
                    </li>
                    <li
                      class="css-0"
                    >
                      コードジェネレータ：Hygen
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div
              class="chakra-accordion__item css-ejn068"
            >
              <button
                aria-controls="accordion-panel-2"
                aria-expanded="false"
                class="chakra-accordion__button css-df84mh"
                data-index="1"
                id="accordion-button-2"
                type="button"
              >
                <div
                  class="css-1eziwv"
                >
                  <h3
                    class="css-1vpdv9u"
                  >
                    TypeScript
                  </h3>
                  <p
                    class="css-1vf9ext"
                  >
                    1.5年
                  </p>
                </div>
                <svg
                  aria-hidden="true"
                  class="chakra-icon chakra-accordion__icon css-186l2rg"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <div
                class="chakra-collapse"
                style="overflow: hidden; display: none; opacity: 0; height: 0px;"
              >
                <div
                  aria-labelledby="accordion-button-2"
                  class="chakra-accordion__panel css-1hb5ode"
                  id="accordion-panel-2"
                  role="region"
                >
                  基本的な型付けからジェネリクスを用いた少し応用的な型付けまで対応可能です。
                </div>
              </div>
            </div>
            <div
              class="chakra-accordion__item css-ejn068"
            >
              <button
                aria-controls="accordion-panel-3"
                aria-expanded="false"
                class="chakra-accordion__button css-df84mh"
                data-index="2"
                id="accordion-button-3"
                type="button"
              >
                <div
                  class="css-1eziwv"
                >
                  <h3
                    class="css-1vpdv9u"
                  >
                    Next.js
                  </h3>
                  <p
                    class="css-1vf9ext"
                  >
                    1.5年
                  </p>
                </div>
                <svg
                  aria-hidden="true"
                  class="chakra-icon chakra-accordion__icon css-186l2rg"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <div
                class="chakra-collapse"
                style="overflow: hidden; display: none; opacity: 0; height: 0px;"
              >
                <div
                  aria-labelledby="accordion-button-3"
                  class="chakra-accordion__panel css-1hb5ode"
                  id="accordion-panel-3"
                  role="region"
                >
                  <a
                    class="chakra-link css-wmzrah"
                    href="https://zenn.dev/a_da_chi/articles/181ea4ccc39580"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    基本的なNext.jsプロジェクトのセットアップ
                    <svg
                      class="chakra-icon css-1pmp75g"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <g
                        fill="none"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-width="2"
                      >
                        <path
                          d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                        />
                        <path
                          d="M15 3h6v6"
                        />
                        <path
                          d="M10 14L21 3"
                        />
                      </g>
                    </svg>
                  </a>
                  から対応可能です。上記の記事以外にも
                  <a
                    class="chakra-link css-wmzrah"
                    href="https://zenn.dev/a_da_chi"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Zenn
                    <svg
                      class="chakra-icon css-1pmp75g"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <g
                        fill="none"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-width="2"
                      >
                        <path
                          d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                        />
                        <path
                          d="M15 3h6v6"
                        />
                        <path
                          d="M10 14L21 3"
                        />
                      </g>
                    </svg>
                  </a>
                  でNext.jsに関する記事を複数執筆しており、累計で500近くのLikeを頂いております。
                </div>
              </div>
            </div>
            <div
              class="chakra-accordion__item css-ejn068"
            >
              <button
                aria-controls="accordion-panel-4"
                aria-expanded="false"
                class="chakra-accordion__button css-df84mh"
                data-index="3"
                id="accordion-button-4"
                type="button"
              >
                <div
                  class="css-1eziwv"
                >
                  <h3
                    class="css-1vpdv9u"
                  >
                    Ruby on Rails
                  </h3>
                  <p
                    class="css-1vf9ext"
                  >
                    2.5年
                  </p>
                </div>
                <svg
                  aria-hidden="true"
                  class="chakra-icon chakra-accordion__icon css-186l2rg"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <div
                class="chakra-collapse"
                style="overflow: hidden; display: none; opacity: 0; height: 0px;"
              >
                <div
                  aria-labelledby="accordion-button-4"
                  class="chakra-accordion__panel css-1hb5ode"
                  id="accordion-panel-4"
                  role="region"
                >
                  モノリシックなアプリケーション開発からAPI開発まで対応可能です。Fat Controllerにならないようにするのはもちろん、デコレータ層やユースケース層を使用してFat Modelにならないように意識しています。テストにはminitestを使用することが多く、日常的にテストを書いています。
                </div>
              </div>
            </div>
            <div
              class="chakra-accordion__item css-ejn068"
            >
              <button
                aria-controls="accordion-panel-5"
                aria-expanded="false"
                class="chakra-accordion__button css-df84mh"
                data-index="4"
                id="accordion-button-5"
                type="button"
              >
                <div
                  class="css-1eziwv"
                >
                  <h3
                    class="css-1vpdv9u"
                  >
                    Go
                  </h3>
                  <p
                    class="css-1vf9ext"
                  >
                    0.1年
                  </p>
                </div>
                <svg
                  aria-hidden="true"
                  class="chakra-icon chakra-accordion__icon css-186l2rg"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <div
                class="chakra-collapse"
                style="overflow: hidden; display: none; opacity: 0; height: 0px;"
              >
                <div
                  aria-labelledby="accordion-button-5"
                  class="chakra-accordion__panel css-1hb5ode"
                  id="accordion-panel-5"
                  role="region"
                >
                  社内ツールの作成で触ったことがある程度です。
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-cx70wh"
      >
        <div
          class="chakra-stack css-3c05cu"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTI4IiBoZWlnaHQ9IjEyOCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Yutaro Adachiのプロフィール画像"
              class="circle-image"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div
            class="chakra-stack css-igigla"
          >
            <h2
              class="css-553wiy"
            >
              Yutaro Adachi
            </h2>
            <p
              class="css-1vf9ext"
            >
              フロントエンドエンジニア
            </p>
          </div>
          <div
            class="chakra-stack css-a9v878"
          >
            <a
              class="chakra-link css-f4h6uy"
              href="https://github.com/yutaroadachi"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="chakra-icon css-1fmr1qf"
                fill="none"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
                />
              </svg>
            </a>
            <a
              class="chakra-link css-f4h6uy"
              href="https://twitter.com/a_da_chi_"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="chakra-icon css-1fmr1qf"
                fill="none"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z"
                />
              </svg>
            </a>
          </div>
        </div>
        <p
          class="css-0"
        >
          大学で経済学を学んだのち、大手SIerでメンバーおよびチームリーダーとして客先常駐でのシステム開発を経験しました。この経験を通して、自分でもコードを書けるようになる必要があると感じ、エンジニアにキャリアチェンジすることを決意しました。
          <br />
          その後スタートアップでエンジニアとしてフロントエンド開発からバックエンド開発まで幅広く経験してきましたが、特にReactを中心としたフロントエンド開発が得意です。また要件定義や設計といった上流工程の経験もしてきたため、一人称で開発に携わることができると自負しております。
        </p>
      </div>
    </div>
  </div>
</div>
`;
